src/OutputWriters/netcdf_output_writer.jl

```diff
@@ -1,135 +1,716 @@
+        Δyᶜᶜᵃ_name = dim_name_generator("dy", grid, c, c, nothing, Val(:y))
+
+        Δyᶠᶠᵃ_field = Field(yspacings(grid, f, f); indices)
+        Δyᶠᶜᵃ_field = Field(yspacings(grid, f, c); indices)
+        Δyᶜᶠᵃ_field = Field(yspacings(grid, c, f); indices)
+        Δyᶜᶜᵃ_field = Field(yspacings(grid, c, c); indices)
+
+        metrics[Δyᶠᶠᵃ_name] = Δyᶠᶠᵃ_field
+        metrics[Δyᶠᶜᵃ_name] = Δyᶠᶜᵃ_field
+        metrics[Δyᶜᶠᵃ_name] = Δyᶜᶠᵃ_field
+        metrics[Δyᶜᶜᵃ_name] = Δyᶜᶜᵃ_field
+    end
+
+    if TZ != Flat
+        Δzᵃᵃᶠ_name = dim_name_generator("dz", grid, nothing, nothing, f, Val(:z))
+        Δzᵃᵃᶜ_name = dim_name_generator("dz", grid, nothing, nothing, c, Val(:z))
+
+        Δzᵃᵃᶠ_field = Field(zspacings(grid, f); indices)
+        Δzᵃᵃᶜ_field = Field(zspacings(grid, c); indices)
+
+        metrics[Δzᵃᵃᶠ_name] = Δzᵃᵃᶠ_field
+        metrics[Δzᵃᵃᶜ_name] = Δzᵃᵃᶜ_field
+    end
+
+    return metrics
 end
 
-const default_rectilinear_dimension_attributes = Dict(
-    "xC"          => Dict("long_name" => "Locations of the cell centers in the x-direction.", "units" => "m"),
-    "xF"          => Dict("long_name" => "Locations of the cell faces in the x-direction.",   "units" => "m"),
-    "yC"          => Dict("long_name" => "Locations of the cell centers in the y-direction.", "units" => "m"),
-    "yF"          => Dict("long_name" => "Locations of the cell faces in the y-direction.",   "units" => "m"),
-    "zC"          => Dict("long_name" => "Locations of the cell centers in the z-direction.", "units" => "m"),
-    "zF"          => Dict("long_name" => "Locations of the cell faces in the z-direction.",   "units" => "m"),
+gather_grid_metrics(grid::ImmersedBoundaryGrid, args...) =
+    gather_grid_metrics(grid.underlying_grid, args...)
+
+#####
+##### Gathering of immersed boundary fields
+#####
+
+# TODO: Proper masks for 2D models?
+flat_loc(T, L) = T == Flat ? nothing : L
+
+# For Immersed Boundary Grids (IBG) with a Grid Fitted Bottom (GFB)
+function gather_immersed_boundary(grid::GFBIBG, indices, dim_name_generator)
+    op_mask_ccc = KernelFunctionOperation{Center, Center, Center}(peripheral_node, grid, Center(), Center(), Center())
+    op_mask_fcc = KernelFunctionOperation{Face, Center, Center}(peripheral_node, grid, Face(), Center(), Center())
+    op_mask_cfc = KernelFunctionOperation{Center, Face, Center}(peripheral_node, grid, Center(), Face(), Center())
+    op_mask_ccf = KernelFunctionOperation{Center, Center, Face}(peripheral_node, grid, Center(), Center(), Face())
+
+    return Dict(
+        "bottom_height" => Field(grid.immersed_boundary.bottom_height; indices),
+        "immersed_boundary_mask_ccc" => Field(op_mask_ccc; indices),
+        "immersed_boundary_mask_fcc" => Field(op_mask_fcc; indices),
+        "immersed_boundary_mask_cfc" => Field(op_mask_cfc; indices),
+        "immersed_boundary_mask_ccf" => Field(op_mask_ccf; indices)
+    )
+end
```

Comment:
This function is assuming a `GridFittedBottom`, but it would be nice to extend it to `GridFittedBoundary` (the only difference is that `GridFittedBoundary` doesn't necessary have a bottom height).

Reply:
Yes! This function only dispatches on `GridFittedBottom` (via `GFBIBG`) but I need to also define `gather_immersed_boundary` for `PartialCellBottom` and `GridFittedBoundary`.

Reply:
This is done now and is actually tested in `test_lagrangian_particle_tracking.jl`!

Reply:
nice!